symplyphysics/laws/gravity/maximum_height_of_a_body_thrown_at_an_angle_to_horizon.py

```diff
@@ -0,0 +1,70 @@
+from sympy import (Eq, solve, sin, pi)
+from sympy.physics.units import acceleration_due_to_gravity as earth_free_fall_acceleration
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output, angle_type)
+from symplyphysics.core.expr_comparisons import expr_equals
+from symplyphysics.laws.kinematic import constant_acceleration_movement_is_parabolic as distance_law
+from symplyphysics.laws.kinematic import planar_projection_is_cosine as projection_law
+from symplyphysics.laws.gravity import maximum_movement_time_of_a_body_thrown_at_an_angle_to_horizon as time_law
+
+# Description
+## Let's say we throw the body at an angle to the horizon with some initial velocity.
+## Then the time of movement (time of fall) of the body will depend only on the initial velocity,
+## the angle of the throw and the acceleration of free fall.
+## The angle of the throw is the angle between the initial velocity vector and the horizontal axis.
+
+## Law is: t = 2 * v0 * sin(a) / g, where
+## t - movement time,
+## v0 - initial velocity,
+## a - angle of throw,
+## g - acceleration of free fall.
+
+height = Symbol("height", units.length)
+
+initial_velocity = Symbol("initial_velocity", units.velocity)
+angle = Symbol("angle", angle_type)
+
+law = Eq(height, initial_velocity**2 * sin(angle)**2 / (2 * earth_free_fall_acceleration))
+
+# This law might be derived via "constant_acceleration_movement_is_parabolic" law, "planar_projection_is_cosine" law
+# and "maximum_movement_time_of_a_body_thrown_at_an_angle_to_horizon" law.
+
+# The law seeks a projection on the horizontal axis, but a projection on the vertical axis is necessary,
+# so the angle is represented as a "pi/2 - angle".
+projection_law_applied = projection_law.law.subs({
+    projection_law.vector_length: initial_velocity,
+    projection_law.vector_angle: (pi / 2) - angle,
+})
+projection_derived = solve(projection_law_applied, projection_law.projection, dict=True)[0][projection_law.projection]
+
+time_law_applied = time_law.law.subs({
+    time_law.initial_velocity : initial_velocity,
+    time_law.angle: angle,
+})
+time_derived = solve(time_law_applied, time_law.movement_time, dict=True)[0][time_law.movement_time]
```

Comment:
You've just made a law to calculate this. Maybe reuse it?

Reply:
> You've just made a law to calculate this. Maybe reuse it?

So I'm overusing the law "maximum_movement_time_of_a_body_thrown_at_an_angle_to_horizon" here.
Is it necessary to reuse it in some other way?

Reply:
Sorry, I've missed it